api/v1/prefix_types.go

```diff
@@ -24,6 +24,7 @@ import (
 // NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
 
 // PrefixSpec defines the desired state of Prefix
+// +kubebuilder:validation:XValidation:rule="!has(oldSelf.site) || has(self.site)", message="Site is required once set"
```

Comment:
Out of curiosity, can you help me understand why we need to do validation here and then once more at `Site` on Line 37? :)

Reply:
Certainly! The first check, at Line 27, ensures that the field actually exists in the data. The second check, at Line 37, ensures that the field is not an empty string. The second step is only performed if the field exists, so if the first validation isn't present the second check could be skipped.